Modules/_functoolsmodule.c

```diff
@@ -1073,6 +1073,7 @@ lru_cache_clear_list(lru_list_elem *link)
 static void
 lru_cache_dealloc(lru_cache_object *obj)
 {
+    _PyObject_GC_UNTRACK(obj);
```

Comment:
`_PyObject_GC_UNTRACK()` shouldn't be used in types that can be subclassed because it can't be called repeatedly. Otherwise the subclass's destructor would need to call `_PyObject_GC_TRACK()` before calling parent's destructor:
```
static void
lru_cache_subclass_dealloc(lru_cache_subclass_object *obj)
{
    _PyObject_GC_UNTRACK(obj);
    Py_XDECREF(obj->foo);
    Py_XDECREF(obj->bar);
    _PyObject_GC_TRACK(obj);
    lru_cache_dealloc((lru_cache_object *)obj);
}
```
This looks too fragile.